include/mata/nfa/nfa.hh

```diff
@@ -205,6 +205,8 @@ public:
 
     BoolVector get_useful_states() const;
 
+    BoolVector get_useful_states_tarjan() const;
```

Comment:
Also some comment what this function returns would be nice (I assume BoolVector for states representing which are useful).

Reply:
I agree that we should replace the original function with the Tarjan one. Furthermore, if we want to keep the classical approach, we should rename it accordingly (something like `get_useful_states_classic()`, `get_useful_states_naive()`, etc.).

Reply:
I remove the original function (and a couple of other obsolete code related to trim) later in this PR. The original code is left basically for Lukas for a comparison (there was some discussion whether it could be simplified somehow according to the original function).

Reply:
Then what about creating a dispatcher function and hiding the implementation details in a separate source file where we can experiment with different implementations?

> I remove the original function ... later in this PR. The original code is left basically for Lukas for a comparison ...

I do not understand. The original code is still there, but are do we think we will ever use it again? If no, we can remove the code entirely. And ideally, we should think about the potential simplification as a part of this PR, before merging it. Or do you mean that the original code is not removed yet and will be before merging the PR? Then it would be OK with me.

Reply:
Fine, I remove it now. Lukas can find the original version in the git history.